docs/DevGuide/Parallelization.md

```diff
@@ -275,18 +302,119 @@ during the next phase. Typically the `execute_next_phase` function should just
 call `start_phase(phase)` on the parallel component. In the future
 `execute_next_phase` may be removed.
 
+## 3. Examples {#dev_guide_parallelization_component_examples}
+
 An example of a singleton parallel component is:
 \snippet Test_AlgorithmParallel.cpp singleton_parallel_component
 
 An example of an array parallel component is:
 \snippet Test_AlgorithmParallel.cpp array_parallel_component
-Elements are inserted into the array by using the Charm++ `insert` member
-function of the CProxy for the array. The `insert` function is documented in
-the Charm++ manual. In the above Array example `array_proxy` is a `CProxy` and
-so all the documentation for Charm++ array proxies applies. SpECTRE always
-creates empty arrays with the constructor and requires users to insert however
-many elements they want and on which cores they want them to be placed. Note
-that load balancing calls may result in array elements being moved.
+
+There are some parallel components that are common to many executables.
+
+- Parallel::GlobalCache (a Parallel::Algorithms::Nodegroup)
+- Parallel::MutableGlobalCache (a Parallel::Algorithms::Group)
+- DgElementArray (a Parallel::Algorithms::Array)
+- observers::Observer (a Parallel::Algorithms::Group)
+- observers::ObserverWriter (a Parallel::Algorithms::Nodegroup)
+
+The MutableGlobalCache deserves special mention, which is why is has its own
+section with instructions on how to use it. See [Mutable items in the
+GlobalCache](#dev_guide_parallelization_mutable_global_cache).
+
+## 4. Placement {#dev_guide_parallelization_component_placement}
+
+The user has some control over where parallel components get placed on the
+resources it is running on. Here is a figure that illustrates how one may place
+parallel components.
+
+\image html charm_node_structure.png "Parallel component placement."
+
+In this example we are running on three (3) nodes that have four (4) cores each.
+For all our executables, we reserve one core of each node purely for
+communication purposes. Nothing else is run on this core. Because of this, what
+Charm++ calls a node, doesn't correspond to a full node on a supercomputer. A
+charm-node simply corresponds to a collection of cores that need not be on the
+same physical node. In our case, a charm-node is represented by the remaining
+cores on a node not used for communication (i.e. the first charm-node
+corresponds to cores 1-3 on the first physical node). Also the definition of a
+charm-core doesn't necessarily have to correspond to an actual core, however,
+for our purposes, it does.
```

Comment:
maybe add `(it could correspond to a hyperthreaded virtual core)`? I think virtualization beyond that doesn't really make sense from a practical point of view